src/app/view/editor-main-view/data-views/trainrunsections.view.ts

```diff
@@ -2147,6 +2348,133 @@ export class TrainrunSectionsView {
     return retPath;
   }
 
+  /**
+   * Detect collapsed node chains and calculate direct visualization path
+   *
+   * A collapsed chain occurs when intermediate nodes are hidden from view.
+   * Example: A-B*-C*-D where B* and C* are collapsed -> create direct path A→D
+   *
+   * Algorithm:
+   * 1. Walk backward from current section to find chain start (non-collapsed node)
+   * 2. Walk forward from current section to find chain end (non-collapsed node)
+   * 3. If start != end, calculate direct path using legacy routing system
+   *
+   * @param currentSection The section that might be part of a collapsed chain
+   * @returns Direct path Vec2D[] if chain detected, null otherwise
+   */
+  private getCollapsedChainPath(currentSection: TrainrunSection): Vec2D[] | null {
+    const trainrunId = currentSection.getTrainrunId();
+
+    const startSection = this.findChainBoundary(currentSection, trainrunId, "backward");
+    const endSection = this.findChainBoundary(currentSection, trainrunId, "forward");
+
+    if (startSection.getId() !== endSection.getId()) {
+      const startNode = startSection.getSourceNode();
+      const endNode = endSection.getTargetNode();
+
+      return TrainrunSectionsView.calculateDirectPath(
+        startNode,
+        endNode,
+        [startSection, endSection],
+        currentSection,
+      );
+    }
+
+    return null;
+  }
+
+  /**
+   * Updates the TrainrunSection's path if it's part of a collapsed chain.
+   * This ensures the visual rendering uses the correct collapsed path instead of the direct path.
+   */
+  private updateTrainrunSectionPathForCollapsedChain(section: TrainrunSection): void {
+    const collapsedPath = this.getCollapsedChainPath(section);
+    if (collapsedPath) {
+      (section as any).pathVec2D = collapsedPath;
```

Comment:
Would be nice to avoid overriding private `TrainrunSection` model fields. I'd like to investigate if we can replace all `TrainrunSection.getPath()` calls with `TrainrunSectionViewObject.getPath()` from https://github.com/OpenRailAssociation/netzgrafik-editor-frontend/pull/605.

Reply:
Done in #605